client/boot/common.js

```diff
@@ -437,10 +433,55 @@ const boot = ( currentUser, registerRoutes ) => {
 	} );
 };
 
-export const bootApp = ( appName, registerRoutes ) => {
+function waitForCookieAuth( user ) {
+	const timeoutMs = 1500;
+	const loggedIn = user.get() !== false;
+	const ipc = require( 'electron' ).ipcRenderer;
+
+	const promiseTimeout = ( ms, promise ) => {
+		const timeout = new Promise( ( _, reject ) => {
+			const id = setTimeout( () => {
+				clearTimeout( id );
+				reject( `Request timed out in ${ ms } ms` );
+			}, ms );
+		} );
+
+		return Promise.race( [ promise, timeout ] );
+	};
+
+	const renderPromise = () => {
+		return new Promise( function ( resolve ) {
+			const sendUserAuth = () => {
+				debug( 'Sending user info to desktop...' );
+				ipc.send( 'user-auth', user, getToken() );
+			};
+
+			if ( loggedIn ) {
+				debug( 'Desktop user logged in, waiting on cookie authentication...' );
+				ipc.on( 'cookie-auth-complete', function () {
+					debug( 'Desktop cookies set, rendering main layout...' );
+					resolve();
+				} );
+				// Send user auth and wait on cookie-auth-complete
+				sendUserAuth();
+			} else {
+				debug( 'Desktop user logged out, rendering main layout...' );
+				// Send user auth and resolve immediately
+				sendUserAuth();
+				resolve();
+			}
+		} );
+	};
+
+	return promiseTimeout( timeoutMs, renderPromise() );
+}
+
+export const bootApp = async ( appName, registerRoutes ) => {
 	const user = userFactory();
-	user.initialize().then( () => {
-		debug( `Starting ${ appName }. Let's do this.` );
-		boot( user, registerRoutes );
-	} );
+	await user.initialize();
+	if ( config.isEnabled( 'desktop' ) ) {
+		await waitForCookieAuth( user );
```

Comment:
If waiting for the `cookie-auth-complete` event times out, then `waifForCookieAuth` returns a rejected promise and the `await` throws an exception. That avoids the entire boot sequence, nothing is ever rendered, and adding the timeout didn't really improve the user's situation.

It would be better to time out gracefully: return a resolved promise and continue the boot. There wont be auth cookies, the stats image won't load, but that's all: everything else in the app will work.